### DISCUSSION TITLE
delete participant crash

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#4204] [Support]
When deleting participant, an EDP message received at the same time which causes crash. 

12-18 11:49:06.683 23214 23271 I fastdds : Reader Proxy removed: 01.0f.66.cc.c8.1a.d9.5c.01.00.00.00|0.0.2.4 -> Function matched_reader_remove
12-18 11:49:06.694 23214 23256 D fastdds : delete_participant
12-18 11:49:06.701 23214 23271 I fastdds : Writer proxy 01.0f.66.cc.c8.1a.d9.5c.01.00.00.00|0.0.5.3 removed from 0.0.2.4 -> Function matched_writer_remove
12-18 11:49:07.959  1301  1301 I Zygote  : Process 23214 exited due to signal 11 (Segmentation fault)

#00 pc 00000000004573cc  libfastdds.so (eprosima::fastrtps::rtps::StatefulReader::matched_writer_remove(eprosima::fastrtps::rtps::GUID_t const&, bool)+2256) (BuildId: e11db706a94621185db1953564e52360)
12-18 11:49:07.872 32541 32541 F DEBUG   :       #01 pc 0000000000393660  libfastdds.so (eprosima::fastrtps::rtps::EDP::unpairWriterProxy(eprosima::fastrtps::rtps::GUID_t const&, eprosima::fastrtps::rtps::GUID_t const&, bool)+768) (BuildId: e11db706a94621185db1953564e52360)
12-18 11:49:07.872 32541 32541 F DEBUG   :       #02 pc 00000000003bce90  libfastdds.so (eprosima::fastrtps::rtps::PDP::remove_remote_participant(eprosima::fastrtps::rtps::GUID_t const&, eprosima::fastrtps::rtps::ParticipantDiscoveryInfo::DISCOVERY_STATUS)+1244) (BuildId: e11db706a94621185db1953564e52360)
12-18 11:49:07.872 32541 32541 F DEBUG   :       #03 pc 00000000003bd6c4  libfastdds.so (eprosima::fastrtps::rtps::PDP::check_remote_participant_liveliness(eprosima::fastrtps::rtps::ParticipantProxyData*)+596) (BuildId: e11db706a94621185db1953564e52360)
12-18 11:49:07.872 32541 32541 F DEBUG   :       #04 pc 00000000003be2b4  libfastdds.so (std::__1::__function::__func<eprosima::fastrtps::rtps::PDP::add_participant_proxy_data(eprosima::fastrtps::rtps::GUID_t const&, bool, eprosima::fastrtps::rtps::ParticipantProxyData const*)::$_0, std::__1::allocator<eprosima::fastrtps::rtps::PDP::add_participant_proxy_data(eprosima::fastrtps::rtps::GUID_t const&, bool, eprosima::fastrtps::rtps::ParticipantProxyData const*)::$_0>, bool ()>::operator()()+20) (BuildId: e11db706a94621185db1953564e52360)

**Comment by advancenXX** [comment]
kindly ping...

**Comment by JesusPoderoso** [comment]
Hi @advancenXX, thanks for using Fast DDS.
In this case, It would be easier to try to solve your problem if you [open an issue ticket](https://github.com/eProsima/Fast-DDS/issues/new?assignees=&labels=triage&projects=&template=bug_report.yml) and provide a reproducer.
